[DOC] fix argument order between n_interpolate and consensus

mscheltienne opened the pull request:
While giving a shot to the `AutoReject` class, I got confused by the docstring not matching the actual arguments. 

I changed the docstring to match the argument order.

TODO:
- [x] Fix argument order/docstring for public `AutoReject` class. [Fixed]
- [x] Argument `thresh_func=None` is not documented in the dosctring. What is it? [Unused]
- [x] Public `AutoReject` class argument order is `n_interpolate=None, consensus=None` while private `_AutoReject` class argument order is `consensus=0.1, n_interpolate=0`. Do you want to do something about this? [Fixed]

jasmainak commented:
Thanks for giving a shot at this!

Here is the documentation for `thresh_func` : https://github.com/autoreject/autoreject/blob/master/autoreject/autoreject.py#L469-L471. 

Regarding the order, no strong feelings but if you do need to change, it must be done in the private class

mscheltienne commented:
Just noticed... `thresh_func` is not used: https://github.com/autoreject/autoreject/blob/master/autoreject/autoreject.py#L889-L904
It does remind me of something.. ;)

Anyway, I'll at least check the argument order later, to at least harmonize it between the public and private class.
I'll change the order in the private class to avoid breaking any user code using positional arguments and I'll check if the private class is called somewhere with positional arguments.

jasmainak commented:
LOL, probably that's why it's not documented. Remember this is originally research code. We intended to use it somehow but forgot to remove it afterwards. It's probably safer to leave it in now to avoid breaking user code

mscheltienne commented:
Added couple of commits to reorder/harmonize private `_AutoReject` arguments. The class is not called anywhere with positional arguments, so nothing should break.

- `dots` argument is not in the docstring. I have no idea what it is, so I can not add it.
- `method` argument, which I renamed to `thresh_method` for consistency is not used either.. https://github.com/autoreject/autoreject/blob/master/autoreject/autoreject.py#L501-L517

mscheltienne commented on autoreject/autoreject.py:
```diff
@@ -50,7 +50,7 @@ def validation_curve(epochs, y=None, param_name="thresh", param_range=None,
     epochs : instance of mne.Epochs.
         The epochs.
-    y : array | None
+    y : array | None
```
That was an auto-modification by my IDE, I didn't pay attention. I don't see the change, do you?

jasmainak replied:
Nopes, it's fine. You can leave it as is

jasmainak commented:
`dots` no need to add in docstring because it's private API. It's a tuple with the `cross_dots` and `self_dots` that are [precomputed for faster interpolation](https://github.com/autoreject/autoreject/blob/97d401c6af1ea870fdc56ca49345dec06833af74/autoreject/utils.py#L395).

mscheltienne commented:
For future development purposes, and since this private class is already well documented, I fixed a typo and added a quick docstring for `dots` to `_AutoReject`. 

I understand this is originally research code, and it really depends on how much of a finish product you want to turn this library into. For me, this kind of mistakes do not help to make me confident in `autoreject`. 

I tried adding `thresh_method` to the attributes of `_AutoReject` (to fix the unused passed argument), but it would have probably break things as the argument is sometimes passed via a partially initialized `thresh_func`.

Anyway, that will be it for this PR. It's just some docstring improvements. Nothing will break ;)

jasmainak commented:
CircleCI fails :-( Think I need to ping @hoechenberger because it seems related to `openneuro-py`

mscheltienne commented:
That's a funny one considering I installed and used the latest version of `openneuro-py` without any problem a few minutes ago.. I fixed the warning issued for the too-short title underline for a title in the new workflow tutorial in this PR.

jasmainak commented:
yeah ... I don't know what's going on either

jasmainak commented:
I can reproduce by the way:

```py
PATTERN=plot_autoreject_workflow make html_dev-pattern
```

This is on a python 3.9 env

hoechenberger commented:
I'll look into this asap (meaning, later afternoon CEST)

hoechenberger commented:
I have released `openneuro-py` `2021.10.1`, which hopefully fixed this issue, and have restarted Circle. Let's hope it passes this time.

jasmainak commented:
Thanks @mscheltienne for the doc fixes !
